From: Janosch Frank <frankja@linux.vnet.ibm.com>
Subject: KVM: s390: instruction-execution-protection support
Patch-mainline: v4.11-rc1
Git-commit: cd1836f583d78bdd15ef748f4d85bf007569c7ad
References: LTC#162428, bsc#1073069

Summary:     kernel: Enable Instruction Execution Protection facility KVM usage
Description: The new Instruction Execution Protection facility
             provides operating systems with the ability to mark pages
             no-execute. This patch also lets KVM guest use this
             feature.

Upstream-Description:

             KVM: s390: instruction-execution-protection support

             The new Instruction Execution Protection needs to be enabled before
             the guest can use it. Therefore we pass the IEP facility bit to the
             guest and enable IEP interpretation.

             Signed-off-by: Janosch Frank <frankja@linux.vnet.ibm.com>
             Reviewed-by: David Hildenbrand <dahi@linux.vnet.ibm.com>
             Signed-off-by: Christian Borntraeger <borntraeger@de.ibm.com>


Signed-off-by: Janosch Frank <frankja@linux.vnet.ibm.com>
Acked-by: Johannes Thumshirn <jthumshirn@suse.de>
---
 arch/s390/kvm/kvm-s390.c |    3 +++
 1 file changed, 3 insertions(+)

--- a/arch/s390/kvm/kvm-s390.c
+++ b/arch/s390/kvm/kvm-s390.c
@@ -1412,6 +1413,8 @@ int kvm_arch_vcpu_setup(struct kvm_vcpu
 		vcpu->arch.sie_block->ecb |= 0x10;
 
 	vcpu->arch.sie_block->ecb2  = 8;
+	if (test_kvm_facility(vcpu->kvm, 130))
+		vcpu->arch.sie_block->ecb2 |= 0x20;
 	vcpu->arch.sie_block->eca   = 0xC1002000U;
 	if (sclp.has_siif)
 		vcpu->arch.sie_block->eca |= 1;

